src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -976,10 +972,8 @@ public static unsafe IntPtr StringToBSTR(string? s)
 
             IntPtr bstr = AllocBSTR(s.Length);
 
-            fixed (char* firstChar = s)
-            {
-                string.wstrcpy((char*)bstr, firstChar, s.Length + 1);
-            }
+            Buffer.Memmove(ref *(char*)bstr, ref s.GetRawStringData(), (nuint)s.Length + 1);
```

Comment:
I am wondering whether it would look better to just use Span in these higher level methods, like:
 
`s.AsSpan().CopyTo(new Span<char>((void*)ptr, s.Length));`

This change would also require fixing AllocBSTR on Unix to zero-terminate the memory chunk. I think it may be actually a bug that AllocBSTR does not zero-terminate the memory chunk on Unix.

Reply:
Maybe we can replace the two usages of AllocBSTR with AllocBSTRByLen (which does the zero-termination) and delete AllocBSTR.